system/ee/ExpressionEngine/Addons/comment/mod.comment.php

```diff
@@ -1887,6 +1886,9 @@ public function subscriber_list()
      */
     public function comment_subscribe()
     {
+        if (ee()->input->get('token') != CSRF_TOKEN) {
```

Comment:
This is checking for `'token'` GET variable, however I don't see where it's set - can you point me ? (again, it would be preferred to not pass that openly, but to pass that as header or POST variable)

Reply:
After taking another look at this, I think I'm fine with the CSRF being in the query string. Previously, these links were completely unsecured and changing them to a header or POST would require refactoring how the Subscribe and Unsubscribe mechanism works. I think the CSRF is a step in the right direction but we would need to decide to refactor this code entirely if we don't want to go with this modified approach. @matthewjohns0n Do you have concerns about this from a security stand-point?